x-pack/legacy/plugins/apm/server/lib/metrics/by_agent/java/gc/fetchAndTransformGcMetrics.ts

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { idx } from '@kbn/elastic-idx';
+import theme from '@elastic/eui/dist/eui_theme_light.json';
+import { Setup } from '../../../../helpers/setup_request';
+import { getMetricsDateHistogramParams } from '../../../../helpers/metrics';
+import { ChartBase } from '../../../types';
+import { getMetricsProjection } from '../../../../../../common/projections/metrics';
+import { mergeProjection } from '../../../../../../common/projections/util/merge_projection';
+import {
+  SERVICE_AGENT_NAME,
+  LABELS,
+  METRIC_JAVA_GC_COUNT,
+  METRIC_JAVA_GC_TIME
+} from '../../../../../../common/elasticsearch_fieldnames';
+
+const colors = [
+  theme.euiColorVis0,
+  theme.euiColorVis1,
+  theme.euiColorVis2,
+  theme.euiColorVis3,
+  theme.euiColorVis4,
+  theme.euiColorVis5,
+  theme.euiColorVis6
+];
+
+export async function fetchAndTransformGcMetrics({
+  setup,
+  serviceName,
+  serviceNodeName,
+  chartBase,
+  fieldName
+}: {
+  setup: Setup;
+  serviceName: string;
+  serviceNodeName?: string;
+  chartBase: ChartBase;
+  fieldName: typeof METRIC_JAVA_GC_COUNT | typeof METRIC_JAVA_GC_TIME;
+}) {
+  const { start, end, client } = setup;
+
+  const projection = getMetricsProjection({
+    setup,
+    serviceName,
+    serviceNodeName
+  });
+
+  const params = mergeProjection(projection, {
+    body: {
+      size: 0,
+      query: {
+        bool: {
+          filter: [
+            ...projection.body.query.bool.filter,
+            { exists: { field: fieldName } },
+            { term: { [SERVICE_AGENT_NAME]: 'java' } }
+          ]
+        }
+      },
+      aggs: {
+        per_pool: {
+          terms: {
+            field: `${LABELS}.name`
+          },
+          aggs: {
+            average: {
+              avg: {
+                field: fieldName
+              }
+            },
+            over_time: {
+              date_histogram: getMetricsDateHistogramParams(start, end),
+              aggs: {
+                max: {
+                  max: {
+                    field: fieldName
+                  }
+                },
+                derivative: {
+                  derivative: {
+                    buckets_path: 'max'
+                  }
+                },
```

Comment:
Maybe add a comment about how the java agent sends gc.count and gc.count as monotonically increasing counters, and that we need to get the delta. 

And perhaps also explain the bucket script below.